questionpy_sdk/webserver/templates/attempt.html.jinja2

```diff
@@ -4,6 +4,10 @@
 {% block head %}
     {{ super() }}
     <script type="module" src="static/script.js"></script>
+    <script src="static/attempt.js"></script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.7/require.js"
+        integrity="sha512-H/RK9lhgLZE7IvypfHj5iUX0fnbaz5gA8y81NQ8F6azabccQuFAVeQdvOYDeAvAsl/WZTOGphkwhhlpCJi157A=="
+        crossorigin="anonymous" referrerpolicy="no-referrer"></script>
```

Comment:
JavaScript von externen Anbietern einzubinden halte ich für problematisch und unnötig.

- Datenschutz: CDNs können Verbindungen unkontrolliert an Staaten außerhalb der EU weiterleiten. Das ist ein Datenschutzproblem und ist im Zweifel nicht DSGVO-konform.
- Setzt aktive Internet-Verbindung vorraus (z.B. jemand der im Zug arbeitet o.ä. ist außen vor).
- Security: Legt die Verantworung effektiv in die Hand eines externen Anbieters. Kompromittierte CDNs hat es schon gegeben...
- Downtimes: Es hat in der Vergangenheit CDN-Downtimes gegeben.

Im Allgmeinen auch unnötig, da es Alternativen gibt:
  - **RequireJS gar nicht erst nutzen** (siehe anderen Kommentar)
  - Die JS-Datei direkt ins Projekt schreiben (wäre hier m.M.n. die pragmatische Lösung, wenn an RequireJS festgehalten wird)
  - (NPM verwenden) - kommt ja wahrscheinlich eh bald mit dem neuen Frontend

Reply:
Ich bin auch kein Fan von CDNs, aber vor dem Hintergrund der angehenden kompletten JS-Umstrukturierung ist das eine pragmatische Zwischenlösung. Später sollte das beim Bundlen direkt mitgeliefert werden.

Bezüglich RequireJS habe ich eingangs mehrere Gründe genannt. Ich denke es bietet eine große Flexibilität für dynamische Abhängigkeiten. Dagegen sind ES6 und die Importmap leider viel zu statisch und andere moderne Alternativen für genau unseren Anwendungsfall mit den Fragepaketen sehe ich nicht.